### PR TITLE
fix(Search): type definition for layout renderer

### LINKED ISSUE #4308: Type definition for Search Category Layout Renderer is incorrect
## Bug Report

https://github.com/Semantic-Org/Semantic-UI-React/blob/61a08733941fb60d6ff09f30c3e2fc13e40448da/src/modules/Search/SearchCategory.js#L25

↓Actually, the argument of this function is a single object, but it is written as two objects in types.
https://github.com/Semantic-Org/Semantic-UI-React/blob/61a08733941fb60d6ff09f30c3e2fc13e40448da/src/modules/Search/SearchCategory.d.ts#L29-L39

↓ Actually, The argument of this function is a single object, but it is written as two objects in JSDoc.
https://github.com/Semantic-Org/Semantic-UI-React/blob/61a08733941fb60d6ff09f30c3e2fc13e40448da/src/modules/Search/Search.d.ts#L61-L68

### Expected

Both should be
```ts
/** 
 * Renders the SearchCategory layout. 
 * 
 * @param {object} props - The SearchCategoryLayout props object.
 * @returns {*} - Renderable SearchCategory layout. 
 */ 
(props: SearchCategoryLayoutProps) => React.ReactElement<any>
```

### Version
2.0.4

### PR DESCRIPTION
**Pull request by bhbs**
Fixes #4308 

- Corrected Types.
- `SearchCategoryLayoutProps` is defined and should be used.

### CODE REVIEW

**Comment by bhbs** [comment]
@layershifter 
Hi, could you review this?
Thank you for nice library.

File: src/modules/Search/SearchCategory.d.ts [hunk 1]
```diff
@@ -29,14 +30,10 @@ export interface StrictSearchCategoryProps {
...
-   * @param {object} categoryContent - The Renderable SearchCategory contents.
-   * @param {object} resultsContent - The Renderable SearchResult contents.
+   * @param {object} props - The SearchCategoryLayout props object.
    * @returns {*} - Renderable SearchCategory layout.
    */
-  layoutRenderer?: (
-    categoryContent: React.ReactElement<any>,
-    resultsContent: React.ReactElement<any>,
-  ) => React.ReactElement<any>
+  layoutRenderer?: (props: SearchCategoryLayoutProps) => React.ReactElement<any>
```
Comment (layershifter) [inline src/modules/Search/SearchCategory.d.ts]:
```suggestion
  layoutRenderer?: (props: Pick<SearchCategoryLayoutProps, 'categoryContent' | 'resultsContent'>) => React.ReactElement<any>
```

We don't pass all props there only `categoryContent` & `resultsContent`, so let's make it more accurate 😉

File: src/modules/Search/Search.d.ts [hunk 2]
```diff
@@ -61,11 +62,10 @@ export interface StrictSearchProps {
   /**
    * Renders the SearchCategory layout.
    *
-   * @param {object} categoryContent - The Renderable SearchCategory contents.
-   * @param {object} resultsContent - The Renderable SearchResult contents.
+   * @param {object} props - The SearchCategoryLayout props object.
    * @returns {*} - Renderable SearchCategory layout.
    */
-  categoryLayoutRenderer?: (props: SearchCategoryProps) => React.ReactElement<any>
+  categoryLayoutRenderer?: (props: SearchCategoryLayoutProps) => React.ReactElement<any>
```
Comment (layershifter) [inline src/modules/Search/Search.d.ts]:
```suggestion
  categoryLayoutRenderer?: (props: Pick<SearchCategoryLayoutProps, 'categoryContent' | 'resultsContent'>) => React.ReactElement<any>
```

Same as in other comment :)

File: src/modules/Search/SearchCategory.d.ts (hunk 1 above)
Comment (bhbs) [reply src/modules/Search/SearchCategory.d.ts]: You're right! Thank you! 😀

**Pull request merged**